src/serac/physics/utilities/finite_element_vector.hpp

```diff
@@ -0,0 +1,218 @@
+     */
+    mfem::Ordering::Type ordering = mfem::Ordering::byVDIM;
+    /**
+     * @brief The name of the field encapsulated by the state object
+     */
+    std::string name = "";
+    /**
+     * @brief A bool denoting if the grid function is managed by sidre
+     * @note This should only be true if calling a constructor from the StateManager class
+     */
+    bool managed_by_sidre = false;
+  };
+
+  /**
+   * @brief Main constructor for building a new finite element vector
+   * @param[in] mesh The problem mesh (object does not take ownership)
+   * @param[in] options The options specified, namely those relating to the order of the problem,
+   * the dimension of the FESpace, the type of FEColl, the DOF ordering that should be used,
+   * and the name of the field
+   */
+  FiniteElementVector(mfem::ParMesh& mesh, Options&& options = {.order            = 1,
+                                                                .vector_dim       = 1,
+                                                                .coll             = {},
+                                                                .ordering         = mfem::Ordering::byVDIM,
+                                                                .name             = "",
+                                                                .managed_by_sidre = false});
+
+  /**
+   * @brief Minimal constructor for a FiniteElementVector given an already-existing field
+   * @param[in] mesh The problem mesh (object does not take ownership)
+   * @param[in] gf The field for the state to create (object does not take ownership)
+   * @param[in] name The name of the field
+   */
+  FiniteElementVector(mfem::ParMesh& mesh, mfem::ParGridFunction& gf, const std::string& name = "");
+
+  /**
+   * @brief Minimal constructor for a FiniteElementVector given a finite element space
+   * @param[in] mesh The problem mesh (object does not take ownership)
+   * @param[in] space The space to use for the finite element state. This space is deep copied into the new FE state
+   * @param[in] name The name of the field
+   */
+  FiniteElementVector(mfem::ParMesh& mesh, mfem::ParFiniteElementSpace& space, const std::string& name = "");
+
+  /**
+   * @brief Returns the MPI communicator for the state
+   * @return The underlying MPI communicator
+   */
+  MPI_Comm comm() const { return detail::retrieve(space_).GetComm(); }
+
+  /**
+   * @brief Returns a non-owning reference to the internal mesh object
+   * @return The underlying mesh
+   */
+  mfem::ParMesh& mesh() { return mesh_; }
+
+  /**
+   * @brief Returns a non-owning reference to the internal FESpace
+   * @return The underlying finite element space
+   */
+  mfem::ParFiniteElementSpace& space() { return detail::retrieve(space_); }
+  /// \overload
+  const mfem::ParFiniteElementSpace& space() const { return detail::retrieve(space_); }
+
+  /**
+   * @brief Returns a non-owning reference to the vector of true DOFs
```

Comment:
minor: I think reference variables in C++ are always "non-owning", so this description is a little bit redundant

since this is a class an analyst might interact with directly, it might be good to link to explain (or link to a document that does) what a "true" dofs are